compiler/docgen.nim

```diff
@@ -886,48 +914,49 @@ proc genItem(d: PDoc, n, nameNode: PNode, k: TSymKind, docFlags: DocFlags) =
   if k == skType and nameNode.kind == nkSym:
     d.types.strTableAdd nameNode.sym
 
-proc genJsonItem(d: PDoc, n, nameNode: PNode, k: TSymKind): JsonNode =
+proc genJsonItem(d: PDoc, n, nameNode: PNode, k: TSymKind): JsonItem =
   if not isVisible(d, nameNode): return
   var
     name = getName(d, nameNode)
     comm = genRecComment(d, n)
     r: TSrcGen
   initTokRender(r, n, {renderNoBody, renderNoComments, renderDocComments})
-  result = %{ "name": %name, "type": %($k), "line": %n.info.line.int,
-                 "col": %n.info.col}
+  result.json = %{ "name": %name, "type": %($k), "line": %n.info.line.int,
```

Comment:
in future work, IMO we should use jsonutils here, doing a similar simplification as was done in https://github.com/nim-lang/Nim/pull/18100, which allows more type-safety (via a type that represents the spec of what's serialized, making deserialization + serialization easier)